onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -4236,25 +4236,24 @@ TEST_F(GraphTransformationTests, FilterEnabledOptimizers) {
 }
 
 TEST_F(GraphTransformationTests, PropagateCastOpsTests) {
+  using Strategy = GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy;
   struct PropagateCastOpsTestSpecs {
     PathString model_uri;
     // Expected number of casts after the transformation with different stratigies and optimization levels
-    std::map<std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int>, int> casts_count_map;
+    std::map<std::pair<Strategy, int>, int> casts_count_map;
     vector<std::string> allow_ops = {};  // Allowed ops for PropagateCastOps graph transformer
   };
-  std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int> insertAndReduce0 = std::make_pair(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::InsertAndReduce, 0);
-  std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int> floodFill1 = std::make_pair(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::FloodFill, 1);
-  std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int> floodFill2 = std::make_pair(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::FloodFill, 2);
-  std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int> floodFill1Plus = std::make_pair(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::FloodFill |
-                                                                                                                             GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::RemoveInputOutputUpDownCasts,
-                                                                                                                         1);
-  std::pair<GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy, int> floodFill2Plus = std::make_pair(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy::FloodFill |
```

Comment:
instead of removing the test cases for RemoveInputOutputUpDownCasts, can we keep them and test passing in the allow list instead?

Reply:
These test cases specific the RemoveInputOutputUpDownCasts because we removed the function, they are not necessary.

Reply:
there can be benefit in retaining them and making sure we get the same functionality as RemoveInputOutputUpDownCasts as with passing allow_list- since thats what we are expecting- to achieve the same result with a different existing functionality.

Reply:
We are only removing the test entry because the strategy is no longer available. I didn't remove any onnx model or test case.
floodFill1Plus and floodFill2Plus are only test scenarios that use the RemoveInputOutputUpDownCasts.

Reply:
The test case using fp16 allowed list is already there.

Reply:
got it, thanks